libs/client-api/src/http_template.rs

```diff
@@ -0,0 +1,232 @@
+use client_api_entity::{
+  AccountLink, CreateTemplateCategoryParams, CreateTemplateCreatorParams,
+  GetTemplateCategoriesQueryParams, GetTemplateCreatorsQueryParams, TemplateCategories,
+  TemplateCategory, TemplateCategoryType, TemplateCreator, TemplateCreators,
+  UpdateTemplateCategoryParams, UpdateTemplateCreatorParams,
+};
+use reqwest::Method;
+use shared_entity::response::{AppResponse, AppResponseError};
+use uuid::Uuid;
+
+use crate::Client;
+
+fn template_api_prefix(base_url: &str) -> String {
+  format!("{}/api/template-center", base_url)
+}
+
+fn category_resources_url(base_url: &str) -> String {
+  format!("{}/category", template_api_prefix(base_url))
+}
+
+fn category_resource_url(base_url: &str, category_id: Uuid) -> String {
+  format!("{}/{}", category_resources_url(base_url), category_id)
+}
+
+fn template_creator_resources_url(base_url: &str) -> String {
+  format!("{}/creator", template_api_prefix(base_url))
+}
+
+fn template_creator_resource_url(base_url: &str, creator_id: Uuid) -> String {
+  format!(
+    "{}/{}",
+    template_creator_resources_url(base_url),
+    creator_id
+  )
+}
+
+impl Client {
+  pub async fn create_template_category(
```

Comment:
since there's many param, maybe can just have client to pass in `CreateTemplateCategoryParams` instead. this can also potentially avoid unneeded `to_string()`.